libqtile/core/manager.py

```diff
@@ -483,18 +507,25 @@ def register_widget(self, w):
             self.widgets_map[w.name] = w
 
     @property
-    def current_layout(self):
+    def current_layout(self) -> Layout:
         return self.current_group.layout
 
     @property
-    def current_group(self):
+    def current_group(self) -> _Group:
         return self.current_screen.group
 
     @property
-    def current_window(self):
+    def current_window(self) -> Optional[base.Window]:
         return self.current_screen.group.current_window
 
-    def reserve_space(self, reserved_space, screen):
+    def reserve_space(
+        self,
+        reserved_space: List[int],  # [left, right, top, bottom]
```

Comment:
It might be good to have this be `Tuple[int, int, int, int]`, since it is required to have length 4. Similarly below.

Reply:
Yep sounds good. I've updated these tuples in a few places, but had to add an ignore comment due to https://github.com/python/mypy/issues/7509. I've put an inline comment beside it so that in the future it can be checked and removed when mypy resolves this.